### PR TITLE
[MM-53249] Allow server to hide or customize the Forgot Password link

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
Mobile app changes for allowing servers to customize the Forgot Password link.

Related server/web app PR: https://github.com/mattermost/mattermost/pull/23831

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53249

```release-note
NONE
```

### CODE REVIEW

**Review by larkox — APPROVED** [review]

File: app/screens/login/form.tsx [hunk 1]
```diff
@@ -8,6 +8,7 @@ import {Keyboard, TextInput, TouchableOpacity, View} from 'react-native';
 import Button from 'react-native-button';
 
 import {login} from '@actions/remote/session';
+import {tryOpenURL} from '@app/utils/url';
```
Comment (enahum) [inline app/screens/login/form.tsx:11]:
```suggestion
import {tryOpenURL} from '@utils/url';
```

and fix the sort

File: app/screens/login/form.tsx [hunk 2]
```diff
@@ -334,7 +340,7 @@ const LoginForm = ({config, extra, serverDisplayName, launchError, launchType, l
                 endAdornment={endAdornment}
             />
 
-            {(emailEnabled || usernameEnabled) && (
+            {(emailEnabled || usernameEnabled) && config.PasswordEnableForgotLink === 'true' && (
```
Comment (enahum) [inline app/screens/login/form.tsx:343]: `PasswordEnableForgotLink` is a new setting present in newer servers right ? if the answer is YES, then previous server versions won't have the ability to select the option

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/screens/login/form.tsx (hunk 2 above)
Comment (devinbinnie) [reply app/screens/login/form.tsx:343]: Ahh, good catch. I'll make sure to just check that it's not explicitly `false`

**Comment by enahum** [comment]
@devinbinnie eslint is going to fail, the imports must be sorted

**Comment by devinbinnie** [comment]
> @devinbinnie eslint is going to fail, the imports must be sorted

Sorry, done.

**Review by abhijit-singh — APPROVED** [review]
Unblocking this assuming it works find since it is straightforward design wise.

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**